src/pages/studyView/StudyViewPageStore.ts

```diff
@@ -4886,6 +5009,85 @@ export class StudyViewPageStore
         return this.genomicDataCountPromises[chartMeta.uniqueKey];
     }
 
+    public getMutationEventChartDataCount(
+        chartMeta: ChartMeta
+    ): MobxPromise<MultiSelectionTableRow[]> {
+        if (
+            !this.mutationDataCountPromises.hasOwnProperty(chartMeta.uniqueKey)
+        ) {
+            this.mutationDataCountPromises[chartMeta.uniqueKey] = remoteData<
+                MultiSelectionTableRow[]
+            >({
+                await: () => [this.selectedSamples],
+                invoke: async () => {
+                    let res: MultiSelectionTableRow[] = [];
```

Comment:
put this all inside a helper function in a util file

Reply:
I extracted them out to StudyViewUtils.tsx